prereise/gather/winddata/hrrr/calculations.py

```diff
@@ -98,34 +99,42 @@ def calculate_pout(wind_farms, start_dt, end_dt, directory):
     dts = [
         dt for dt in pd.date_range(start=start_dt, end=end_dt, freq="H").to_pydatetime()
     ]
-    data = np.empty((len(dts), len(wind_farm_to_closest_wind_grid_indices)))
+    # Fetch wind speed data for each wind farm (or store NaN as applicable)
+    wind_speed_data = np.empty((len(dts), len(wind_farm_to_closest_wind_grid_indices)))
     for i, dt in tqdm(enumerate(dts)):
         gribs = pygrib.open(formatted_filename(dt))
-        u_component = gribs.select(name=U_COMPONENT_SELECTOR)[0].values.flatten()
-        v_component = gribs.select(name=V_COMPONENT_SELECTOR)[0].values.flatten()
-        wind_farm_specific_u_component = u_component[
-            wind_farm_to_closest_wind_grid_indices
-        ]
-        wind_farm_specific_v_component = v_component[
-            wind_farm_to_closest_wind_grid_indices
-        ]
-        wind_farm_specific_magnitude = np.sqrt(
-            pow(wind_farm_specific_u_component, 2)
-            + pow(wind_farm_specific_v_component, 2)
-        )
-        power = np.array(
-            [
-                get_power(
-                    turbine_power_curves,
-                    state_power_curves,
-                    wind_farm_specific_magnitude[j],
-                    turbine_types[j],
-                )
-                for j in range(wind_farm_ct)
+        try:
+            u_component = gribs.select(name=U_COMPONENT_SELECTOR)[0].values.flatten()
+            v_component = gribs.select(name=V_COMPONENT_SELECTOR)[0].values.flatten()
+            wind_farm_specific_u_component = u_component[
+                wind_farm_to_closest_wind_grid_indices
             ]
-        )
-        data[i] = power
-
-    df = pd.DataFrame(data=data, index=dts, columns=wind_farms.index)
+            wind_farm_specific_v_component = v_component[
+                wind_farm_to_closest_wind_grid_indices
+            ]
+            wind_speed_data[i] = np.sqrt(
+                pow(wind_farm_specific_u_component, 2)
+                + pow(wind_farm_specific_v_component, 2)
+            )
+        except ValueError:
+            # If the GRIB file is empty, no wind speed values can be selected
+            wind_speed_data[i] = np.nan
+
+    # For each column, linearly interpolate any NaN values
+    linear(wind_speed_data)
+    # Then calculate wind power based on wind speed
+    wind_power_data = [
+        [
+            get_power(
+                turbine_power_curves,
+                state_power_curves,
+                wind_speed_data[i, j],
+                turbine_types[j],
+            )
+            for j in range(wind_farm_ct)
+        ]
+        for i, _ in tqdm(enumerate(dts))
```

Comment:
Do we need the enumerate here? It seems we just need the length of the `dts`

Reply:
`enumerate()` vs. `range(len())`, which is more pythonic?

Reply:
`for i in tqdm(dts)`?

Reply:
> 

I think we will need the index rather than the element of `dts` here.

Reply:
`tqdm(dts, total=len(dts))` would work, but not sure if that's better than `enumerate`/`range`. It's unclear to me if any is the most pythonic, but I think `enumerate` is fine

Reply:
We could also refactor how we build `wind_speed_data`--as a dataframe instead of a numpy array--and then we could build `wind_power_data` using `apply` calls instead of list comprehensions.

Reply:
See https://github.com/Breakthrough-Energy/PreREISE/pull/221#discussion_r708509549.

Reply:
Aha. Maybe I'll leave this alone for now then.

Reply:
Done. I tried to strike a good balance between compactness and readability.